### PR TITLE
feat(manager/custom): allow packageName instead of depName

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Refactors regex customer manager so that packageName can be extracted instead of depName, meaning depName is no longer mandatory.

## Context

Transition to packageName-first extraction.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/custom/regex/readme.md [hunk 1]
```diff
@@ -88,15 +88,13 @@ But you don't want to write a regex custom manager rule for _each_ variable.
 Instead you enhance your `Dockerfile` like this:
 
 ```Dockerfile
-ARG IMAGE=node:12@sha256:6e5264cd4cfaefd7174b2bc10c7f9a1c2b99d98d127fc57a802d264da9fb43bd
-FROM ${IMAGE}
```
Comment (rarkins) [inline lib/modules/manager/custom/regex/readme.md:92]: This type of pattern is now handled natively by the `dockerfile` manager so is not a good example for regex manager

File: lib/modules/manager/custom/regex/readme.md [hunk 2]
```diff
@@ -29,7 +29,7 @@ Before Renovate can look up a dependency and decide about updates, it needs this
 Configuration-wise, it works like this:
 
 - You must capture the `currentValue` of the dependency in a named capture group
-- You must have either a `depName` capture group or a `depNameTemplate` config field
+- You must have either a `depName` capture group or a `depNameTemplate` config field, or `packageName` or `packageNameTemplate`
 - You can optionally have a `packageName` capture group or a `packageNameTemplate` if it differs from `depName`
```
Comment (viceice) [inline lib/modules/manager/custom/regex/readme.md:33]: I think this needs to be removed?

File: lib/modules/platform/gerrit/scm.ts [hunk 3]
```diff
@@ -34,7 +34,7 @@ export class GerritScm extends DefaultGitScm {
       .findChanges(repository, searchConfig, true)
       .then((res) => res.pop());
     if (change) {
-      return change.current_revision! as LongCommitSha;
+      return change.current_revision as LongCommitSha;
```
Comment (viceice) [inline lib/modules/platform/gerrit/scm.ts:37]: unintentional?

File: lib/modules/manager/custom/regex/readme.md (hunk 2 above)
Comment (rarkins) [inline lib/modules/manager/custom/regex/readme.md:33]:
```suggestion
```

File: lib/modules/platform/gerrit/scm.ts (hunk 3 above)
Comment (rarkins) [reply lib/modules/platform/gerrit/scm.ts]: This diff should have disappeared once merged from main

Comment (rarkins) [reply lib/modules/platform/gerrit/scm.ts]: https://github.com/renovatebot/renovate/pull/28833

File: docs/usage/configuration-options.md [hunk 4]
```diff
@@ -689,7 +689,7 @@ The `regex` manager which is based on using Regular Expression named capture gro
 You must have a named capture group matching (e.g. `(?<depName>.*)`) _or_ configure its corresponding template (e.g. `depNameTemplate`) for these fields:
 
 - `datasource`
-- `depName`
+- `depName` or `packageName` (either or both is OK)
```
Comment (secustor) [inline docs/usage/configuration-options.md:692]:
```suggestion
- `depName` and / or `packageName`
```

File: lib/modules/manager/custom/regex/readme.md [hunk 5]
```diff
@@ -29,8 +29,7 @@ Before Renovate can look up a dependency and decide about updates, it needs this
 Configuration-wise, it works like this:
 
 - You must capture the `currentValue` of the dependency in a named capture group
-- You must have either a `depName` capture group or a `depNameTemplate` config field
-- You can optionally have a `packageName` capture group or a `packageNameTemplate` if it differs from `depName`
+- You must have either a `depName` capture group or a `depNameTemplate` config field, or `packageName` or `packageNameTemplate`
```
Comment (secustor) [inline lib/modules/manager/custom/regex/readme.md:32]:
```suggestion
- You must have either a `depName` or `packageName` capture group, or use on of the respective template fields (  `depNameTemplate` and `packageNameTemplate` )
```

File: lib/modules/manager/custom/regex/readme.md [hunk 6]
```diff
@@ -88,15 +87,13 @@ But you don't want to write a regex custom manager rule for _each_ variable.
...
-# renovate: datasource=github-releases depName=composer/composer
+# renovate: datasource=github-tags depName=node packageName=nodejs/node versioning=node
+ENV NODE_VERSION=20.10.0
+# renovate: datasource=github-releases depName=composer packageName=composer/composer
 ENV COMPOSER_VERSION=1.9.3
-# renovate: datasource=docker depName=docker versioning=docker
+# renovate: datasource=docker packageName=docker versioning=docker
 ENV DOCKER_VERSION=19.03.1
-# renovate: datasource=npm depName=yarn
+# renovate: datasource=npm packageName=yarn
```
Comment (secustor) [inline lib/modules/manager/custom/regex/readme.md:96]:
```suggestion
# renovate: datasource=npm depName=YARN packageName=yarn
```
Example of using both.

**Review by secustor — CHANGES_REQUESTED** [review]

File: lib/modules/manager/custom/regex/readme.md (hunk 6 above)
Comment (rarkins) [reply lib/modules/manager/custom/regex/readme.md:96]: There's already two examples earlier so I preferred to have this one without

**Review by secustor — APPROVED** [review]

**Comment by latin-1** [comment]
https://github.com/renovatebot/renovate/blob/main/lib/config/presets/internal/custom-managers.ts

Should we replace all `depNameTemplate` with `packageNameTemplate` in this file?

**Pull request merged**